### PR TITLE
Feat: Added generator for middleware

### PR DESCRIPTION
**Pull request by adampash**
Addresses #256

### CODE REVIEW

**Comment by zacharygolba** [comment]
I'm getting the following error due to the file path for the newly created middleware being invalid:

```
{ Error: ENOENT: no such file or directory, open '/Users/zacharygolba/Desktop/blog/app/middlewares/set-cors-headers.js'
    at Error (native)
  cause:
   { Error: ENOENT: no such file or directory, open '/Users/zacharygolba/Desktop/blog/app/middlewares/set-cors-headers.js'
       at Error (native)
     errno: -2,
     code: 'ENOENT',
     syscall: 'open',
     path: '/Users/zacharygolba/Desktop/blog/app/middlewares/set-cors-headers.js' },
  isOperational: true,
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/Users/zacharygolba/Desktop/blog/app/middlewares/set-cors-headers.js' }
```

I'm guessing this is due to the `type` parameter being pluralized when it should within the generate command handler.

File: src/packages/cli/commands/generate.js [hunk 1]
```diff
@@ -49,13 +50,18 @@ export async function generateType(type, name, cwd, attrs = []) {
     case 'controller':
       data = controllerTemplate(name, attrs);
       break;
+
+    case 'middleware':
+      data = middlewareTemplate(name);
+      break;
   }
 
   if (type === 'model') {
```
Comment (zacharygolba) [inline src/packages/cli/commands/generate.js]: Ahh here's the culprit. We should probably replace that with a regexp `/^(model|middleware)$/.test(type)`.

Comment (zacharygolba) [inline src/packages/cli/commands/generate.js]: Never mind that's the name haha. I got excited.

**Comment by zacharygolba** [comment]
Ahh I figured out why I was getting that error. I had to remove the locally installed Lux to get the globally npm linked cli rather than the locally installed one. This is working as expect 👍. Disregard my comments.

**Comment by adampash** [comment]
Ha great! Yeah I was wondering what was going wrong! It does actually give you a sense of some polishing that the generator could use, though — you'll get the same error anytime you misspell a generator. 

Instead, if it doesn't match anything, you might consider, at some point, giving suggestions for what generators are available.

**Comment by zacharygolba** [comment]
> Instead, if it doesn't match anything, you might consider, at some point, giving suggestions for what generators are available.

That's an awesome idea! Would you mind creating an issue for that with a feature label?

**Comment by zacharygolba** [comment]
Added the `needs-update` label to address `lux d middleware <name>` command as discussed on Flowdock and in #262.

**Comment by adampash** [comment]
Added `lux d middleware <name>` in latest commit.

**Comment by zacharygolba** [comment]
👍  Works great!

**Pull request merged**